lib/utils.js

```diff
@@ -135,3 +138,17 @@ export function closest(selector) {
         return null;
     }
 }
+
+
+/**
+ * Returns an array of Minimatch instances to test whether a filename
+ * is ignored. Cache the Minimatch instances instead of re-compiling
+ * the regexp.
+ */
+export function ignoredPatterns() {
+    if (this.minimatches === undefined ) {
```

Comment:
I'm unsure what `this` is set to when the function is imported into another file. Is it the global for this module? 

In any case, just adding a `let _minimatches = null;` above the function and using that instead would clear that confusion up.

Reply:
Also, this caching means that changes to the preference wouldn't take effect in existing windows. I can think of two solutions to this:

1. Make the cache and object keyed by the glob patterns, and re-fetch the config each time. That way we always get the current config, but can cache minimatch objects.
2. Add an observer for changes to the config and invalidate the cache when the config changes.

Either would be fine IMO. The former seems easier.